### ISSUE TITLE
Make padding color configurable through parameter

### ISSUE THREAD

**Issue by DeRealMorgan** [ultralytics/ultralytics#23166]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Google LiteRT-support library only supports black padding. Yolo hardcodes gray 114 color for padding. 

Goal is to expose this padding color and make it configurable via a parameter.

### Use case

Padding color should be aligned between Yolo and LiteRT support lib.

Inference on edge devices shall be as efficient as possible. 

Black padding is a result of initializing a default bitmap. If the bitmap has to be colored, preprocessing is less efficient on edge devices (e.g. adds 114ms on a Galaxy S22). 

By setting the same padding color during training, this preprocessing step can be skipped before inference leading to faster overall execution without a performance penalty due to misaligned colors.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @DeRealMorgan, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the detailed feature request about making the letterbox/padding color configurable (to align with Google LiteRT’s black padding) 🧩⚡ This is an automated response to help with triage, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us evaluate this feature request faster, please share a minimal snippet showing your exact preprocessing path and where the padding color is currently applied (Python/CLI), plus the target deployment pipeline details (LiteRT). If you can, also include a small benchmark before/after demonstrating the impact you mentioned 📈

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
You can already make padding black at the augmentation level: `ultralytics.data.augment.LetterBox` has a `padding_value` argument (default `114`) documented in the [LetterBox reference](https://docs.ultralytics.com/reference/data/augment/), but today it isn’t plumbed through as a top-level `yolo train/predict/export` parameter everywhere (and there are a couple other hardcoded `114` pads in the codebase), so your request makes sense for LiteRT alignment.

As a temporary workaround, if you’re OK with a local fork, you can change the default in `ultralytics/data/augment.py` (and retrain) so your exported model matches LiteRT’s black padding:

```python
from ultralytics.data.augment import LetterBox

lb = LetterBox(new_shape=(640, 640), padding_value=0)  # black padding
```

If you can confirm whether your padding is applied only in the Android pre-processing step (outside the model) and which export path you’re using (`tflite` / `onnx` / etc.), that will help us ensure the config is exposed in the right place without breaking existing defaults.

**Comment by DeRealMorgan** [comment]
Yes, I know about the source code change. It will work for now ;).

Yes I am exporting to .tflite, but I also have seen another request, let me try to find it:
https://github.com/ultralytics/ultralytics/issues/19209

**Comment by glenn-jocher** [comment]
Thanks for confirming it’s the `.tflite` path; since a TFLite export doesn’t bake preprocessing into the graph, padding color needs to be kept consistent between your LiteRT letterbox step and Ultralytics’ preprocessing/training pipeline (so this likely needs to be exposed as an end-to-end preprocessing setting, not just an export flag) per the [TFLite export and deployment guide](https://docs.ultralytics.com/integrations/tflite/). If you can paste the exact LiteRT resize/letterbox code path (RGB/BGR, dtype, and normalization), we can make sure the knob lands in the right place and also treat #19209 as related.

**Comment by DeRealMorgan** [comment]
https://github.com/tensorflow/tflite-support/blob/9c16db6ae503bf3fd04f09d43f56b5e14bfb40b6/tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/ops/ResizeWithCropOrPadOp.java

**Comment by glenn-jocher** [comment]
Yep—`ResizeWithCropOrPadOp` in LiteRT does fixed **zero-padding (black)** (and center-crops if the input is larger), so there’s no way to match Ultralytics’ default `114` padding on the LiteRT side; if you can paste the exact `ImageProcessor` chain you’re using around it (the resize op, channel order, and normalization), we can expose `padding_value` end-to-end on the Ultralytics preprocessing/training path without accidentally changing the resize/crop behavior ([ResizeWithCropOrPadOp API docs](https://ai.google.dev/edge/api/tflite/java/org/tensorflow/lite/support/image/ops/ResizeWithCropOrPadOp)).

**Comment by DeRealMorgan** [comment]
I cant provide the chain right now, but I
1. RotateOp
2. resize so the longest side is equal to model input size (e.g. 640) (I think it is called ResizeOp)
3. CropOrPadOp
4. NormalizeOp

**Comment by glenn-jocher** [comment]
That ImageProcessor chain is effectively “resize longest side to `imgsz` + center pad”, so it should match Ultralytics’ `LetterBox` behavior as long as you run it with `auto=False` (no stride rounding) and a black `padding_value=0` during train/val/predict; the two remaining common mismatch sources are the `ResizeOp` method (`BILINEAR` vs `NEAREST_NEIGHBOR`) and the exact `NormalizeOp` parameters (mean/std vs just `/255`, and whether the exported `.tflite` input is float32 or int8), so when you can, please paste those ctor values (or the `yolo export ...` args) and we can plumb `padding_value` end-to-end without changing existing defaults (see the current `LetterBox` implementation in [`ultralytics/data/augment.py`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/data/augment.py#L1509-L1673)).
